test/src/Command/QueueCommandTest.php

```diff
@@ -65,6 +68,31 @@ public function it_will_delete_messages_from_queue_after_processing()
         $this->assertEquals(0, $this->queue->count(), 'Expected an empty queue');
     }
 
+    /**
+     * @test
+     */
+    public function it_will_remove_item_from_queue_if_process_fails()
+    {
+        $application = new Application();
+        $queue_command = $this->getMockBuilder(QueueCommand::class)
```

Comment:
a subclass of `ApplicationSpecificQueueCommand` in this file (`ProcessingErrorQueueCommand`?) could avoid the complex mocking:
```
class ProcessingErrorQueueCommand extends ApplicationSpecificQueueCommand
{
    protected function process(InputInterface $input, QueueItem $item, $entity = null)
    {
        throw new Exception('Fail gracefully'));
    }
}
```

Reply:
I considered that. But because the logger is used multiple times it was easier to catch the one call to the error method by mocking the LoggerInterface.

Reply:
Scratch that. I’ll take a look again tomorrow morning and perhaps refactor.

Reply:
@giorgiosironi I have refactored. Please review.